File: app/src/main/java/com/boringdroid/systemui/TaskInfo.java
package com.boringdroid.systemui;

import android.content.ComponentName;
import android.graphics.drawable.Drawable;

public class TaskInfo {
    private int mId;
    private ComponentName mBaseActivityComponentName;
    private ComponentName mRealActivityComponentName;
    private String mPackageName;
    private Drawable mIcon;

    public int getId() {
        return mId;
    }

    public String getPackageName() {
        return mPackageName;
    }

    public Drawable getIcon() {
        return mIcon;
    }

    public void setBaseActivityComponentName(ComponentName baseActivityComponentName) {
        mBaseActivityComponentName = baseActivityComponentName;
    }

    public void setRealActivityComponentName(ComponentName realActivityComponentName) {
        mRealActivityComponentName = realActivityComponentName;
    }

    public void setPackageName(String packageName) {
        mPackageName = packageName;
    }

    public void setId(int id) {
        mId = id;
    }

    public void setIcon(Drawable icon) {
        mIcon = icon;
    }

    @Override
    public boolean equals(Object another) {
        if (!(another instanceof TaskInfo)) {
            return false;
        }
        TaskInfo task = (TaskInfo) another;
        // The task id is unique in system.
        return mId == task.mId;
    }

    @Override
    public int hashCode() {
        return mId;
    }

    @Override
    public String toString() {
        return "Task id " + mId + ", origin " + mBaseActivityComponentName
                + ", real " + mRealActivityComponentName + ", package " + mPackageName;
    }
}

File: app/src/main/java/com/boringdroid/systemui/TaskInfo.kt
package com.boringdroid.systemui

import android.content.ComponentName
import android.graphics.drawable.Drawable

class TaskInfo {
    var id = 0
    private var baseActivityComponentName: ComponentName? = null
    private var realActivityComponentName: ComponentName? = null
    var packageName: String? = null
    var icon: Drawable? = null

    fun setBaseActivityComponentName(baseActivityComponentName: ComponentName?) {
        this.baseActivityComponentName = baseActivityComponentName
    }

    fun setRealActivityComponentName(realActivityComponentName: ComponentName?) {
        this.realActivityComponentName = realActivityComponentName
    }

    override fun equals(other: Any?): Boolean {
        if (other !is TaskInfo) {
            return false
        }
        // The task id is unique in system.
        return id == other.id
    }

    override fun hashCode(): Int {
        return id
    }

    override fun toString(): String {
        return (
            "Task id " +
                id +
                ", origin " +
                baseActivityComponentName +
                ", real " +
                realActivityComponentName +
                ", package " +
                packageName
        )
    }
}
